### PR TITLE
Scripting: added ability to generate param docs for lua applets and drivers

### PR DESCRIPTION
**Pull request by tridge**
This allows for parameter docs in lua applets and drivers. This will make using the applets a lot easier for users
we could get rid of the need for // in the lua docs with a bit more regex work
parameter docs added for one script (VTOL-quicktune.lua) as a test

Example output: http://uav.tridgell.net/tmp/Parameters.rst

### CODE REVIEW

File: Tools/autotest/param_metadata/param_parse.py [hunk 1]
```diff
@@ -89,17 +109,13 @@ def find_vehicle_parameter_filepath(vehicle_name):
...
 libraries.append(ap_vehicle_lib)
 
+libraries.append(lua_applets())
+
 error_count = 0
 current_param = None
 current_file = None
 
 
-def debug(str_to_print):
```
Comment (peterbarker) [inline Tools/autotest/param_metadata/param_parse.py:97]: Moved for some reason?

**Review by peterbarker — COMMENTED** [review]
Does it work, 'though?

Sample output?

File: Tools/autotest/param_metadata/param_parse.py (hunk 1 above)
Comment (tridge) [reply Tools/autotest/param_metadata/param_parse.py:97]: yes, used in an earlier place, needed to move

**Comment by tridge** [comment]
> Sample output?

sample added

**Review by peterbarker — APPROVED** [review]
LGTM.

@Hwurzburg these new parameters will be appearing on the Wiki - but most vehicles won't have them.  We might need to work out how to point people at the relevant scripts to get those parameters.

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
I am against this being emitted in Parameters.rst for obvious reasons since a wiki user would have no clue that the parameter is only associated if he used a certain script, assuming they even know what scripting is...I can see the benefit to GCS but its just confusion in the wiki...have to have a tag of some kind to prevent it being emitted in Parameters.rst

**Comment by tridge** [comment]
@Hwurzburg I've changed it to not output lua parameters to the rst file, though I think we should re-consider as having them in the wiki for things like VTOL-quicktune would be good I think

**Comment by Hwurzburg** [comment]
I thought VTOL quicktune was going to be moved to C++ pretty soon....we already direct users to use it in the VTOL tuning section as a preferred tuning method for those autopilots that can run scripting...moving it to C++ would allow the numerous F405 plane controllers to use it also...

**Comment by peterbarker** [comment]
@tridge isn't this going to cause problems with the CI checks which ensure all downloaded parameters exist in your downloaded parameter list?  We might need to fix the CI test to not care about things from LUA applets/drivers

... also, all parameter namespaces will have to be unique in LUA examples/drivers.

**Comment by tridge** [comment]
> isn't this going to cause problems with the CI checks which ensure all downloaded parameters exist in your downloaded parameter list?

only if we do those checks with the lua script added, which I don't think we do

**Comment by tridge** [comment]
> I thought VTOL quicktune was going to be moved to C++ pretty soon

yes, I'd like to do that, but certainly not for 4.4.x. Note that adding params for VTOL-quicktune is just an example, I want to check it all works and displays in the GCS, then add params to the other applets and drivers

**Comment by Hwurzburg** [comment]
@tridge I think that adding an autogenerated parameter page for LUA scripts in the wiki with each script's parameters being titled in a section like this would be good and not confusing:
==========================
LUA Script Generated Parameters
==========================

VTOL-quicktune.lua
===============
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Scripting/applets/VTOL-quicktune.lua
QUIK)_ENABLE
.
.
etc.
plane_aerobatics.lua
===============
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Scripting/applets/Aerobatics/FixedWing/plane_aerobatics.lua

AEROM_ANG_ACCEL .....
.
,
,
etc.

just like we do for firmware params and log messages...should be similar to the existing parameter emit script

**Review by Hwurzburg — APPROVED** [review]

**Pull request merged**